### ISSUE TITLE
💅 `useJsxKeyInIterable` incorrect behavior: Cannot determine whether this child has the required key prop.

### ISSUE THREAD

**Issue by Ttou** [biomejs/biome#2468]
### Environment information

```bash
CLI:
  Version:                      1.7.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.20.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.15.7"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        style/noNonNullAssertion = "off"
                                suspicious/noExplicitAny = "off"

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/useJsxKeyInIterable

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAYgByAGUAYQBkAGMAcgB1AG0AYgBzACAAPQAgAFsAewAgAHQAaQB0AGwAZQA6ACAAIgAxADEAMQAiACwAIABwAGEAdABoADoAIAAiAC8AIgAgAH0AXQA7AAoACgBjAG8AbgBzAHQAIABBAHAAcAAgAD0AIAAoACkAIAA9AD4AIAB7AAoACQByAGUAdAB1AHIAbgAgACgACgAJAAkAPABkAGkAdgA%2BAAoACQAJAAkAewBiAHIAZQBhAGQAYwByAHUAbQBiAHMALgBtAGEAcAAoACgAdgAsACAAaQApACAAPQA%2BAAoACQAJAAkACQBpACAAPQA9AD0AIAB0AGgAaQBzAC4AYgByAGUAYQBkAGMAcgB1AG0AYgBzAC4AbABlAG4AZwB0AGgAIAAtACAAMQAgAD8AIAAoAAoACQAJAAkACQAJADwAcwBwAGEAbgAgAGsAZQB5AD0AewB2AC4AcABhAHQAaAB9AD4AewB2AC4AdABpAHQAbABlAH0APAAvAHMAcABhAG4APgAKAAkACQAJAAkAKQAgADoAIAAoAAoACQAJAAkACQAJADwAZABpAHYAIABrAGUAeQA9AHsAdgAuAHAAYQB0AGgAfQA%2BAHsAdgAuAHQAaQB0AGwAZQB9ADwALwBkAGkAdgA%2BAAoACQAJAAkACQApAAoACQAJAAkAKQB9AAoACQAJADwALwBkAGkAdgA%2BAAoACQApADsACgB9ADsACgA%3D

### Expected result

no error tip

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by EvHaus** [comment]
Seems like `useJsxKeyInIterable` doesn't understart ternaries. Would it difficult to add support for that?

**Comment by Sec-ant** [comment]
I don't know if it's easy to add support for ternaries. But IMO we **shouldn't** emit errors when we cannot determine whether the child has the required key prop.

**Comment by jtannas** [comment]
Here's a different instance of it happening but without a ternary.
https://biomejs.dev/playground/?lintRules=all&code=PABDAGgAZQBjAGsAQgBvAHgATABpAHMAdAA%2BAAoACQB7AGEAbABsAE8AcAB0AGkAbwBuAHMALgBtAGEAcAAoAAoACQAJACgAbwBwAHQAaQBvAG4AKQAgAD0APgAgAG8AcAB0AGkAbwBuAC4AdgBpAHMAaQBiAGwAZQAgACYAJgAgADwAQwBoAGUAYwBrAGIAbwB4ACAAawBlAHkAPQB7AGAAYwBoAGUAYwBrAGIAbwB4AF8AJAB7AG8AcAB0AGkAbwBuAC4AaQBkAH0AYAB9ACAALwA%2BACwACgAJACkAfQAKADwALwBDAG8AbAB1AG0AbgBMAGkAcwB0AD4AOwAKAA%3D%3D

**Comment by ematipico** [comment]
> Here's a different instance of it happening but without a ternary. [biomejs.dev/playground/?lintRules=all&code=PABDAGgAZQBjAGsAQgBvAHgATABpAHMAdAA%2BAAoACQB7AGEAbABsAE8AcAB0AGkAbwBuAHMALgBtAGEAcAAoAAoACQAJACgAbwBwAHQAaQBvAG4AKQAgAD0APgAgAG8AcAB0AGkAbwBuAC4AdgBpAHMAaQBiAGwAZQAgACYAJgAgADwAQwBoAGUAYwBrAGIAbwB4ACAAawBlAHkAPQB7AGAAYwBoAGUAYwBrAGIAbwB4AF8AJAB7AG8AcAB0AGkAbwBuAC4AaQBkAH0AYAB9ACAALwA%2BACwACgAJACkAfQAKADwALwBDAG8AbAB1AG0AbgBMAGkAcwB0AD4AOwAKAA%3D%3D](https://biomejs.dev/playground/?lintRules=all&code=PABDAGgAZQBjAGsAQgBvAHgATABpAHMAdAA%2BAAoACQB7AGEAbABsAE8AcAB0AGkAbwBuAHMALgBtAGEAcAAoAAoACQAJACgAbwBwAHQAaQBvAG4AKQAgAD0APgAgAG8AcAB0AGkAbwBuAC4AdgBpAHMAaQBiAGwAZQAgACYAJgAgADwAQwBoAGUAYwBrAGIAbwB4ACAAawBlAHkAPQB7AGAAYwBoAGUAYwBrAGIAbwB4AF8AJAB7AG8AcAB0AGkAbwBuAC4AaQBkAH0AYAB9ACAALwA%2BACwACgAJACkAfQAKADwALwBDAG8AbAB1AG0AbgBMAGkAcwB0AD4AOwAKAA%3D%3D)

I would argue that this isn't a good case. You use `.map` and conditionally returning item, which is an anti pattern in the most general sense, like, you shouldn't do this in any case. You should first call `.filter`, and then `.map`

**Comment by ematipico** [comment]
> I don't know if it's easy to add support for ternaries. But IMO we **shouldn't** emit errors when we cannot determine whether the child has the required key prop.

We see two alternatives:
1. Use the control flow to determine all paths, and make sure that all do a return
2. Do a dirty check of possible parents, and do not trigger the rule if we are inside certain nodes, e.g. if/else, ternary operator, switch, etc.

We can't use `1.` because we already use the semantic model
